src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -286,231 +405,247 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
+
+        {
+            for (uint256 xi = land.x; xi < land.x + land.size; xi += childQuadSize) {
+                for (uint256 yi = land.y; yi < land.y + land.size; yi += childQuadSize) {
+                    uint256 ownerChild;
+                    bool ownAllIndividual;
+                    if (childQuadSize < 3) {
+                        ownAllIndividual = _checkAndClear(from, _getQuadId(LAYER_1x1, xi, yi)) && ownerOfAll;
+                    } else {
+                        ownAllIndividual = _regroupQuad(
+                            from,
+                            to,
+                            Land({x: xi, y: yi, size: childQuadSize}),
+                            false,
+                            childQuadSize / 2
+                        );
+                        uint256 idChild = _getQuadId(childLayer, xi, yi);
+                        ownerChild = _owners[idChild];
+                        if (ownerChild != 0) {
+                            if (!ownAllIndividual) {
+                                require(ownerChild == uint256(from), "not owner of child Quad");
+                            }
+                            _owners[idChild] = 0;
+                        }
+                    }
+                    ownerOfAll = (ownAllIndividual || ownerChild != 0) && ownerOfAll;
+                }
             }
         }
-        if(set) {
-            if(!ownerOfAll) {
-                require(
-                    _ownerOfQuad(3, x, y) == from,
-                    "not owner of all sub quads nor parent quads"
-                );
+
+        if (set) {
+            if (!ownerOfAll) {
+                require(_ownerOfQuad(land.size, land.x, land.y) == from, "not owner of all sub quads nor parent quads");
             }
             _owners[quadId] = uint256(to);
             return true;
         }
+
         return ownerOfAll;
     }
 
-    function _ownerOfQuad(uint256 size, uint256 x, uint256 y) internal view returns (address) {
-        uint256 layer;
-        uint256 parentSize = size * 2;
+    function _idInPath(
+        uint256 i,
+        uint256 size,
+        uint256 x,
+        uint256 y
+    ) internal pure returns (uint256) {
+        uint256 row = i / size;
+        if (row % 2 == 0) {
+            // allow ids to follow a path in a quad
+            return _getQuadId(LAYER_1x1, (x + (i % size)), (y + row));
+        } else {
+            return _getQuadId(LAYER_1x1, (x + size) - (1 + (i % size)), (y + row));
+        }
+    }
+
+    function _isQuadMinted(
+        Land[] memory mintedLand,
+        Land memory quad,
+        uint256 index
+    ) internal pure returns (bool) {
+        for (uint256 i = 0; i < index; i++) {
+            Land memory land = mintedLand[i];
+            if (land.size > quad.size) {
+                if (quad.x >= land.x && quad.x < land.x + land.size) {
+                    if (quad.y >= land.y && quad.y < land.y + land.size) return true;
+                }
+            }
```

Comment:
The size of the land should be greater than the size of the quad, so that the quad could be inside the land and other two condition are such that the co-ordinates of quad are inside the land.

Reply:
Sure, but why do we need 3 if when one is enough ?

Reply:
As there could be a quads which can be smaller than the land in the landMinted Array but not a part of that land so we need to check all the three conditions.